docs/DEPLOYMENT.md

```diff
@@ -1,32 +1,116 @@
+7. Go to ECR and create a new repository named `plio-backend-staging`.
+8. Now go to ECS and create a new task definition with name `plio-backend-staging`.
+   1. Set the task role as `ecsTaskExecutionRole`.
+   2. Set the task memory and task CPU based on your needs.
+   3. Create a new container with name `plio-backend-staging`.
+   4. Enter port `80` in the port mapping field.
+   5. Use `.env.example` file to set all the required environment variables for your container.
+   6. Save the container definition and the task definition.
+   7. You will see the new task definition within the list.
+9.  Go to clusters and create a new cluster with name `plio-staging-cluster`. (skip this step if you've already created a VPC when setting up backend repository)
+   1. Use `Networking only` option. We will go with serverless deployment so that we don't worry about managing our own server instances.
+   2. Don't create a new VPC for your cluster. We'll use the VPC created in previous step in the next step of creating a service.
+   3. Click on create button.
+   4. You will see the new cluster within the list of clusters.
+10. Get into `plio-staging-cluster` and create a new service.
+   1. Set launch type to Fargate. We'll use serverless deployments for Plio.
+   2. Name the service as `plio-staging-backend`.
+   3. Under task definition, select `plio-backend-staging` and use latest revision.
+   4. Set the number of tasks to be one.
+   5. Service type to be `REPLICA`.
+   6. Minimum healthy percentage should be 100 and maximum percent to be 200. Minimum healthy percentage defines the percentage of tasks that should be running at all times. Maximum percent defines how many additional tasks the service can create in parallel to running tasks, before killing the running tasks.
+   7. Deployment type to be `rolling update`.
+   8. Keep other values as default.
+   9. Use the Cluster VPC and the subnet that you configured with Elastic IP.
+   10. Auto-assign public IP to have `ENABLED`. Otherwise, it makes the task accessible only through VPC and not public.
+   11. Under load balancing, select the Network Load Balancing option and select the `plio-backend-staging` load balancer.
+   12. Inside "Container to Load Balancer", click on Add to load balancer option and select `plio-backend-staging` in the target group.
+   13. For auto-scaling, go with "Do not adjust the service's desired count" for staging.
+   14. Review and create service.
+11. Next, go to your GitHub repository and create a new environment from settings tab.
+    1. Name the environment as `Staging`.
+    2. Make sure you have added the following GitHub secrets on repository level. If not, add these as your environment secrets.
+       - AWS_ACCESS_KEY_ID
+       - AWS_SECRET_ACCESS_KEY
+       - AWS_REGION
+12. Once done, make some changes to the code so that the GitHub workflow `deploy_to_ecs_staging.yml` gets triggered.
```

Comment:
How can we add superuser here? - by directly adding to the DB?

Reply:
Have added a custom Django command `python manage.py pliosuperuser` and added it to entrypoint.sh.